### PR TITLE
add routes for resource pinning

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: concourse/client/api.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
     PipelineConfig,
     ResourceVersion,
     Worker,
+    PipelineResource,
```
Comment (AndreasBurger) [inline concourse/client/api.py:33]: please sort

Comment (RaphaelVogel) [reply concourse/client/api.py]: done

File: concourse/client/api.py [hunk 2]
```diff
@@ -266,6 +274,21 @@ def abort_build(self, build_id):
...
+    def pin_resource_version(self, pipeline_name: str, resource_name: str, resource_version_id: int):
+        url = self.routes.pin_resource_version(pipeline_name, resource_name, resource_version_id)
+        self._put(url, "")
+
+    @ensure_annotations
+    def unpin_resource(self, pipeline_name: str, resource_name: str):
+        url = self.routes.unpin_resource(pipeline_name, resource_name)
+        self._put(url, "")
+
+    def pin_coment(self, pipeline_name: str, resource_name: str, comment: str):
```
Comment (ccwienk) [inline concourse/client/api.py:287]: typo: comment (two m)

Comment (ccwienk) [reply concourse/client/api.py:287]: also: ensure annotations?

File: concourse/client/routes.py [hunk 3]
```diff
@@ -124,6 +128,23 @@ def list_workers(self):
...
+            'pin'
+        )
+
+    @ensure_annotations
+    def unpin_resource(self, pipeline_name: str, resource_name: str):
+        return self._api_url('pipelines', pipeline_name, 'resources', resource_name, 'unpin')
+
+    @ensure_annotations
+    def pin_coment(self, pipeline_name: str, resource_name: str):
+        return self._api_url('pipelines', pipeline_name, 'resources', resource_name, 'pin_comment')
```
Comment (ccwienk) [inline concourse/client/routes.py:146]: consider adding a "pin_with_comment" convenience method

**Pull request merged**